lib/full-site-editing/block-templates.php

```diff
@@ -93,18 +112,62 @@ function _gutenberg_get_template_files( $template_type ) {
 				// Subtract ending '.html'.
 				-5
 			);
-			$template_files[] = array(
+
+			$new_template_item = array(
 				'slug'  => $template_slug,
 				'path'  => $template_file,
 				'theme' => $theme_slug,
 				'type'  => $template_type,
 			);
+
+			if ( 'wp_template_part' === $template_type ) {
+				$template_files[] = _gutenberg_conditionally_add_template_part_section_info(
+					$new_template_item,
+					$template_part_data
+				);
+			} else {
+				$template_files[] = $new_template_item;
+			}
 		}
 	}
 
 	return $template_files;
 }
 
+/**
+ * Attempts to read the theme.json and return the 'template-parts' field.
+ *
+ * @return array Template part data from theme.json or empty array if not found.
+ */
+function _gutenberg_get_template_part_info_from_theme_json() {
+	if ( is_readable( locate_template( 'experimental-theme.json' ) ) ) {
+		$theme_json = file_get_contents( locate_template( 'experimental-theme.json' ) );
+		$data       = json_decode(
+			$theme_json,
+			true
+		);
+	}
+	if ( isset( $data['template-parts'] ) ) {
+		return $data['template-parts'];
+	}
+	return array();
```

Comment:
Should we add newline before return here to make it consistent with the rest of the formatting? (same goes for if block above and a couple of other changes in this PR)